src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/ContextHolder.cs

```diff
@@ -42,7 +45,63 @@ public unsafe byte[] GetBytes()
         Span<byte> byteSpan = MemoryMarshal.AsBytes(structSpan);
         return byteSpan.ToArray();
     }
-    public IPlatformAgnosticContext Clone() => new CotnextHolder<T>() { Context = Context };
+    public IPlatformAgnosticContext Clone() => new ContextHolder<T>() { Context = Context };
     public void Clear() => Context = default;
     public void Unwind(Target target) => Context.Unwind(target);
+
+    public bool TrySetRegister(Target target, string fieldName, TargetNUInt value)
+    {
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        switch (field.FieldType)
+        {
+            case Type t when t == typeof(ulong) && target.PointerSize == sizeof(ulong):
+                field.SetValueDirect(__makeref(Context), value.Value);
+                return true;
+            case Type t when t == typeof(uint) && target.PointerSize == sizeof(uint):
+                field.SetValueDirect(__makeref(Context), (uint)value.Value);
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public bool TryReadRegister(Target target, string fieldName, out TargetNUInt value)
+    {
+        value = default;
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        if (field.GetValue(Context) is not object fieldValue) return false;
```

Comment:
This is kind of a weird check. It seems like the actual function of it is to check whether the field value is null, in which case a clearer way to write it would be
```csharp
object fieldValue = field.GetValue(Context);
if (fieldValue == null) return false;
```
I was actually surprised to learn that `x is object y` would evaluate to `false` for `null`.
It's also redundant with the ulong and uint checks below, so I think what you can actually just write is:
```csharp
object fieldValue = field.GetValue(Context);
```
and the two ulong/uint checks below will do the rest of the work, falling through to the `return false` at the bottom.